### PR TITLE
96 add a gfp vae surrogate modeling task

### PR DESCRIPTION
**Pull request by RMichae1**
1. added CBas objective function: SequenceGP model, named "gp"
2. added CBas objective function: VAE ELBO, named "elbo"
3. added CBas objective function: VAE embedding, named "vae"
4. included GP model class and persisted .npy assets
5. corrected data loading with shuffling if seed is present

### CODE REVIEW

File: src/poli/objective_repository/gfp_cbas/assets/models/vae/vae_0_decoder_weights_5k_1.h5
Comment (miguelgondu) [inline src/poli/objective_repository/gfp_cbas/assets/models/vae/vae_0_decoder_weights_5k_1.h5:1]: `.h5`s are already included in the `setup.cfg`, so no worries on this side!

File: src/poli/objective_repository/gfp_cbas/cbas_wrapper.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def decode(self, z, grad=False):
         else:
             return self._decoder(z)
 
-    def transform_sequences_to_atoms(self, x: [torch.Tensor]) -> torch.Tensor:
+    def transform_sequences_to_atoms(self, x: torch.Tensor) -> torch.Tensor:
         """
         Turns a sequence into an approximately Dirac distribution around the sequence.
         :param x:
```
Comment (miguelgondu) [inline src/poli/objective_repository/gfp_cbas/cbas_wrapper.py:82]: We're currently following [`numpy` standards for documentation](https://numpydoc.readthedocs.io/en/latest/format.html). But don't worry about this! I'll change them later on.

**Review by miguelgondu — COMMENTED** [review]
I left some comments! Small summary:
1. One important bit is to add `*.npy` to the `setup.cfg` file.
2. There are a couple of files that I think were tracked without intention, or maybe I'm not following :(.

I haven't cloned locally nor tried to reproduce, though. I'll do that later on!

File: src/poli/core/chemistry/tdc_problem_factory.py
[the remark this answers is not in the document]
Comment (RMichae1) [reply src/poli/core/chemistry/tdc_problem_factory.py:1]: Looks like the rebase wasn't clean and the file should have been removed. Fixed that.

File: src/poli/objective_repository/gfp_cbas/cbas_wrapper.py (hunk 1 above)
Comment (RMichae1) [reply src/poli/objective_repository/gfp_cbas/cbas_wrapper.py:82]: Ah, I see. I've changed it to comply with my VSC recommendations. Can also change my syntax checker.

**Pull request merged**